src/models/room.ts

```diff
@@ -1897,35 +1897,57 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         this.threadsReady = true;
     }
 
+    /**
+     * Calls {@link processPollEvent} for a list of events.
+     *
+     * @param events - List of events
+     */
     public async processPollEvents(events: MatrixEvent[]): Promise<void> {
-        const processPollStartEvent = (event: MatrixEvent): void => {
-            if (!M_POLL_START.matches(event.getType())) return;
+        for (const event of events) {
```

Comment:
MT added this early return as a performance improvement to the similar beacon code, could be good here https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/models/room-state.ts#L500

Reply:
Added [here](https://github.com/matrix-org/matrix-js-sdk/pull/3246/files#diff-1252c1ee1e945dedde8e3e6e96f4b29013a7fc3cd8b83fabc762f2d0add859a8R1909). Can you have another look @kerryarchibald ?

Reply:
Looks good, thanks